app-shared/src/androidJvmMain/kotlin/io/github/droidkaigi/confsched/component/OneStepForwardController.kt

```diff
@@ -0,0 +1,42 @@
+package io.github.droidkaigi.confsched.component
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.snapshotFlow
+import androidx.compose.runtime.snapshots.SnapshotStateList
+import androidx.navigation3.runtime.NavKey
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+
+@Composable
+internal fun rememberOneStepForwardControllerForDesktop(
+    backStack: SnapshotStateList<NavKey>,
+): () -> Unit {
+    var redoKey by remember { mutableStateOf<NavKey?>(null) }
+    var prev by remember { mutableStateOf(backStack.toList()) }
+
+    LaunchedEffect(backStack) {
+        snapshotFlow { backStack.toList() }
+            .map { it to prev }
+            .distinctUntilChanged()
+            .collect { (cur, old) ->
+                redoKey = if (cur.size + 1 == old.size && cur == old.dropLast(1)) {
+                    old.last()
+                } else {
+                    null
+                }
+                prev = cur
+            }
+    }
```

Comment:
I think this code could be simplified and made more readable as follows:
```kotlin
LaunchedEffect(backStack) {
    var previous = backStack.toList()

    snapshotFlow { backStack.toList() }
        .distinctUntilChanged()
        .collect { current ->
            redoKey = if (current == previous.dropLast(1)) {
                previous.last()
            } else {
                null
            }
            previous = current
        }
}
```

- `prev` doesn’t need to be exposed outside of `LaunchedEffect`, so we can keep it fully encapsulated.
- Renaming `cur` to `current`, `prev` to `previous` improves clarity.
- The explicit size check (`current.size + 1 == previous.size`) is unnecessary, because the `dropLast(1)` comparison already implies it.